Sources/StreamCore/Logger/Logger.swift

```diff
@@ -130,91 +130,158 @@ public struct LogSubsystem: OptionSet, CustomStringConvertible, Sendable {
 }
 
 public enum LogConfig {
+    struct Configuration {
```

Comment:
Having `Configuration` inside LogConfig sounds confusing. Shall we make it `State`?